en/docs/develop-components/develop-components-with-git.md

```diff
@@ -28,14 +28,44 @@ Once you have connected your Git repository to Choreo, you can build, deploy, an
-|-------------|-----|------|-----------------------------------------------|
-|Issues       | Y   | N    | Read component id label to filter the pull requests|
-|Metadata     | Y   | N    | List repositories                             |
-|Contents     | Y   | Y    | List branches and create a branch to commit sample code|
+|Permission   | Read| Write| Description                                                           |
+|-------------|-----|------|-----------------------------------------------------------------------|
+|Issues       | Y   | N    | Read component ID label to filter the pull requests                   |
+|Metadata     | Y   | N    | List repositories                                                     |
+|Contents     | Y   | Y    | List branches and create a branch to commit sample code               |
 |Pull Request | Y   | Y    | Create a pull request if you have chosen to start with a Choreo sample|
-|Webhooks     | Y   | Y    | Trigger automatic deployment and configuration generation|
+|Webhooks     | Y   | Y    | Trigger automatic deployment and configuration generation             |
 
+### Add Git submodules to a project
+
+Choreo provides [Git submodule](https://git-scm.com/book/en/v2/Git-Tools-Submodules) support when you connect your GitHub repository to Choreo. This allows you to manage and include external repositories effectively within Choreo build pipelines. Key benefits of this capability include:
+
+  - **Code sharing without duplication**: Avoiding code duplication by using submodules to maintain shared libraries across multiple projects, ensuring a single source of truth.
+  - **Efficient third-party library management**: Managing third-party libraries as submodules to update them independently and track changes easily, avoiding direct code integration.
+
+For example, when you [work with the Micro Integrator (MI) runtime in Choreo](./work-with-the-micro-integrator-runtime-in-choreo.md), you can use Git submodules to reuse MI templates and sequences across components without duplication.
+
+!!! tip 
+    If you encounter an error stating that you cannot clone a submodule due to insufficient permissions, follow the instructions below to grant the necessary permissions:
+
+      - For a personal account:
+
+         1. Sign in to your personal GitHub account.
+         2. In the upper-right corner, click your profile picture, and then click **Settings**.
+         3. In the left navigation menu, go to the **Integrations** section and click **Applications**.
+         4. Under the **Installed GitHub Apps** tab, click **Configure** corresponding to **choreo.dev**.
+         5. Under **Repository Access**, grant access to the necessary repositories.
+
+      - For an organization account:
+
+         1. Sign in to your organization's GitHub account.
+         2. In the upper-right corner, click your profile picture, and then click **Settings**.
```

Comment:
This is wrong, this should be the settings tab in the Top navbar, refer below image, i'm not whether it's called a navbar or not :)

<img width="1439" alt="Screenshot 2024-07-31 at 9 53 38 AM" src="https://github.com/user-attachments/assets/35f744cf-ea9a-40ae-a13c-ad2670b0502b">

Reply:
As discussed, updated based on how this is referred to in GitHub docs.